File: SIN BDD/RESTFULL_DOTNET_CONUNI_GR06/RESTFULL_DOTNET_CONUNI_GR06/04. CLICONSOLA/cliente_consola_rest_dotnet_conuni_gr06/src/Servicio/ServicioMasa.cs
using Ec.Edu.Monster.Modelo;
using Ec.Edu.Monster.Utils;

namespace Ec.Edu.Monster.Servicio;

public class ServicioMasa
{
    private readonly ClienteRest cliente = new();

    public Resultado KilogramosALibras(double kilogramos) => cliente.Convertir("masa", "kilogramosALibras", kilogramos);
    public Resultado GramosAOnzas(double gramos) => cliente.Convertir("masa", "gramosAOnzas", gramos);
    public Resultado ToneladasAKilogramos(double toneladas) => cliente.Convertir("masa", "toneladasAKilogramos", toneladas);
    public Resultado LibrasAOnzas(double libras) => cliente.Convertir("masa", "librasAOnzas", libras);
    public Resultado MiligramosAGramos(double miligramos) => cliente.Convertir("masa", "miligramosAGramos", miligramos);
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/04. CLICONSOLA/cliente_consola_restful_java_conuni_gr06/src/ec/edu/monster/modelo/ServicioMasa.java
package ec.edu.monster.modelo;

public class ServicioMasa {

    private final ClienteRest clienteRest = new ClienteRest();

    public double kilogramosALibras(double kilogramos) throws Exception {
        return invocar("/masa/kilogramos-a-libras", kilogramos);
    }

    public double gramosAOnzas(double gramos) throws Exception {
        return invocar("/masa/gramos-a-onzas", gramos);
    }

    public double toneladasAKilogramos(double toneladas) throws Exception {
        return invocar("/masa/toneladas-a-kilogramos", toneladas);
    }

    public double librasAOnzas(double libras) throws Exception {
        return invocar("/masa/libras-a-onzas", libras);
    }

    public double miligramosAGramos(double miligramos) throws Exception {
        return invocar("/masa/miligramos-a-gramos", miligramos);
    }

    private double invocar(String ruta, double valor) throws Exception {
        String respuesta = clienteRest.get(ruta, valor);
        return ClienteRest.extraerNumero(respuesta, "valor");
    }
}
